### ISSUE TITLE
wallets.csv -> wallets.json

### ISSUE THREAD

**Issue by ligi** [ethereum/ethereum-org-website#2730]
I think important wallet properties can change over time (also see #2715) - having a csv storing the data makes things harder to update and maintenance quite difficult.
So I suggest changing from csv to json for more flexilibilty.
If size is a concern here the json could be postprocessed to become a csv again - but this way only algorithms need to touch it - not humans ;)

Also wondering if there could be data-sharing between https://walletconnect.org/wallets and this site. cc @pedrouid

**Comment by pedrouid** [comment]
Totally... WalletConnect's website is actually close to being updated to use the registry which you can use it as well here: https://github.com/walletconnect/walletconnect-registry

**Comment by ligi** [comment]
Oh nice - [that json](https://registry.walletconnect.org/data/wallets.json
)  is actually already quite close to what I had in mind when opening this issue.

**Comment by pedrouid** [comment]
However this registry only includes Dapps and Wallets that support WalletConnect so I'm not sure if you would also need to aggregate some other data of other Ethereum apps

**Comment by ligi** [comment]
Ya - I don't think the data should be taken directly. But if the same format is used it is easier to share/merge/..

**Comment by pedrouid** [comment]
Sounds good. As you can see the data is already being served statically from this url:

https://registry.walletconnect.org/data/wallets.json

Logos can also be fetched by the id in three different sizes ("sm" | "md" | "lg")

https://registry.walletconnect.org/logo/[size]/[id].jpeg

**Comment by samajammin** [comment]
Thanks for raising this @ligi (& @pedrouid for weighing in).

@ligi do you mind expanding on your reasoning here for my understanding? Why is JSON easier to update/maintain vs. CSV? I'm certainly open to switching over, just want to understand why one is better. Seems like a minor implementation detail to me - Gatsby ingests them the same & queries those data file via GraphQL.

**Comment by ligi** [comment]
I see the following advantages:
 * easier correlation between key and value
 * better when it is not simply a (mostly binary) value for a key - but e.g. a set (thinking about e.g. the platform property suggested in #2715)
 * scales better - I think there should be more and more properties added and I think an entry like this is already messy:
 `metamask,MetaMask,https://metamask.io/,,TRUE,FALSE,TRUE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE
`

and with adding more properties it will get even worse. 
Yea it is just a minor detail - but I think it should be done before adding more properties to not end up in a mess.

**Comment by ligi** [comment]
hm - no way for the author to reopen the issue? Do I need to create a new one?

**Comment by minimalsm** [comment]
Converted the current CSV to JSON, didn't have time to get my head around the GraphQL to take this further.

https://pastebin.com/9vvcdj7f

**Comment by ligi** [comment]
Nice! Not sure why you need GraphQL in that context?

**Comment by minimalsm** [comment]
The implementation makes use of GraphQL and I was struggling to get my head around all the parts, will have another look when i'm more fresh.

**Comment by ShubhankarKG** [comment]
Is this still up for grabs? Wanted to know how to proceed on this one then.

**Comment by samajammin** [comment]
Is this complete @corwintines? I believe we solved as part of https://github.com/ethereum/ethereum-org-website/issues/6274 but LMK if I'm wrong.

**Comment by samajammin** [comment]
Perhaps not - we now have this TS file with wallet data:
https://github.com/ethereum/ethereum-org-website/blob/dev/src/data/wallets/wallet-data.ts

But I still see the CSV in there - can this be removed?
https://github.com/ethereum/ethereum-org-website/blob/dev/src/data/wallets.csv

**Comment by corwintines** [comment]
I'll check this out this week and check if we are good to get rid of the csv.

**Comment by corwintines** [comment]
We converted to ts instead of json. I think that is ok to call this issue complete as this came from some codebase upgrades.

**Comment by samajammin** [comment]
Closing this out - solved in #6457 & https://github.com/ethereum/ethereum-org-website/pull/7318

**Comment by samajammin** [comment]
> Also wondering if there could be data-sharing between https://walletconnect.org/wallets and this site

I still think this is a neat idea @ligi so feel free to re-open or create a new issue if you'd like to push that discussion forward!
